### ISSUE TITLE
please provide steps: how to predict for new dataset after finetune?

### ISSUE THREAD

**Issue by divyangashah** [MorinoseiMorizo/jparacrawl-finetune#2]
hi there,

I have finetuned model & generated best*.pt file.
can you describe how to predict for new dataset using this?

Thanks!

**Comment by MorinoseiMorizo** [comment]
Hi @divyangashah ,

I have prepared the example script for prediction.
https://github.com/MorinoseiMorizo/jparacrawl-finetune/blob/master/en-ja/decode.sh
https://github.com/MorinoseiMorizo/jparacrawl-finetune/blob/master/ja-en/decode.sh

You may need to modify some variables like TEST_SRC (at line 11 to 19 of the script).
If you still have a problem please ask me again!

**Comment by divyangashah** [comment]
okay, Thanks. you revert back.
I will check.
Hope it will return translated sentences only and filter out tokenized sentences & probabilities.

Thanks!

**Comment by divyangashah** [comment]
hi @MorinoseiMorizo ,
I am trying to execute part of the script for prediction on my input file using below command:

!python3 /content/fairseq/generate.py $DATA_DIR \
    --gen-subset test \
    --path $MODEL_DIR/average/average.pt \
    --max-tokens 1000 \
    --beam 6 \
    --lenpen 1.0 \
    --log-format simple \
    --remove-bpe \
    | tee $MODEL_DIR/$B.hyp

but, it throws this error:

usage: generate.py [-h] [--no-progress-bar] [--log-interval N]
                   [--log-format {json,none,simple,tqdm}]
                   [--tensorboard-logdir DIR] [--seed N] [--cpu] [--fp16]
                   [--memory-efficient-fp16] [--fp16-no-flatten-grads]
                   [--fp16-init-scale FP16_INIT_SCALE]
                   [--fp16-scale-window FP16_SCALE_WINDOW]
                   [--fp16-scale-tolerance FP16_SCALE_TOLERANCE]
                   [--min-loss-scale D]
                   [--threshold-loss-scale THRESHOLD_LOSS_SCALE]
                   [--user-dir USER_DIR] [--empty-cache-freq EMPTY_CACHE_FREQ]
                   [--all-gather-list-size ALL_GATHER_LIST_SIZE]
                   [--model-parallel-size N]
                   [--checkpoint-suffix CHECKPOINT_SUFFIX]
                   [--quantization-config-path QUANTIZATION_CONFIG_PATH]
                   [--criterion {label_smoothed_cross_entropy,sentence_ranking,composite_loss,cross_entropy,sentence_prediction,masked_lm,nat_loss,adaptive_loss,legacy_masked_lm_loss,binary_cross_entropy,label_smoothed_cross_entropy_with_alignment,vocab_parallel_cross_entropy}]
                   [--tokenizer {moses,space,nltk}]
                   [--bpe {subword_nmt,gpt2,characters,fastbpe,bytes,byte_bpe,sentencepiece,hf_byte_bpe,bert}]
                   [--optimizer {lamb,adagrad,sgd,adadelta,adafactor,adam,adamax,nag}]
                   [--lr-scheduler {cosine,reduce_lr_on_plateau,tri_stage,polynomial_decay,fixed,inverse_sqrt,triangular}]
                   [--task TASK] [--num-workers N]
                   [--skip-invalid-size-inputs-valid-test] [--max-tokens N]
                   [--max-sentences N] [--required-batch-size-multiple N]
                   [--dataset-impl FORMAT] [--data-buffer-size N]
                   [--gen-subset SPLIT] [--num-shards N] [--shard-id ID]
                   [--path FILE] [--remove-bpe [REMOVE_BPE]] [--quiet]
                   [--model-overrides DICT] [--results-path RESDIR] [--beam N]
                   [--nbest N] [--max-len-a N] [--max-len-b N] [--min-len N]
                   [--match-source-len] [--no-early-stop] [--unnormalized]
                   [--no-beamable-mm] [--lenpen LENPEN] [--unkpen UNKPEN]
                   [--replace-unk [REPLACE_UNK]] [--sacrebleu]
                   [--score-reference] [--prefix-size PS]
                   [--no-repeat-ngram-size N] [--sampling]
                   [--sampling-topk PS] [--sampling-topp PS] [--temperature N]
                   [--diverse-beam-groups N] [--diverse-beam-strength N]
                   [--diversity-rate N] [--print-alignment] [--print-step]
                   [--iter-decode-eos-penalty N] [--iter-decode-max-iter N]
                   [--iter-decode-force-max-iter] [--iter-decode-with-beam N]
                   [--iter-decode-with-external-reranker]
                   [--retain-iter-history]
                   [--decoding-format {unigram,ensemble,vote,dp,bs}]
                   [--momentum M] [--weight-decay WD] [--force-anneal N]
                   [--lr-shrink LS] [--warmup-updates N] [-s SRC] [-t TARGET]
                   [--load-alignments] [--left-pad-source BOOL]
                   [--left-pad-target BOOL] [--max-source-positions N]
                   [--max-target-positions N]
                   [--upsample-primary UPSAMPLE_PRIMARY] [--truncate-source]
                   [--eval-bleu] [--eval-bleu-detok EVAL_BLEU_DETOK]
                   [--eval-bleu-detok-args JSON] [--eval-tokenized-bleu]
                   [--eval-bleu-remove-bpe [EVAL_BLEU_REMOVE_BPE]]
                   [--eval-bleu-args JSON] [--eval-bleu-print-samples]
                   **data**
**generate.py: error: the following arguments are required: data**

can you help me how to execute script?
 it will be good you can explain bit more as average model is generated but can't predict on new data.

Thanks!

**Comment by MorinoseiMorizo** [comment]
Hi, 

Did you run the whole script?
The error says the variable $DATA_DIR is not defined, but it should be set by the earlier part of the script.
As written in `decode.sh`, you should run `preprocess.py` first then try `generate.py`.

I recommend you to run the original `decode.sh` example with our pre-trained model and the provided source text before you try your original fine-tuned model and your source file.
The detailed steps are explained in README.md.

If you still have a problem, please feel free to ask!

**Comment by MorinoseiMorizo** [comment]
I will close this issue since two weeks have passed.
If you still have a problem, please feel free to reopen this!
